### PR TITLE
Get the unit tests working, and clean up the tests and fixtures

### PR DESCRIPTION
**Pull request by karalekas**
Relevant tests should now be skipped if the compiler and QVM are not available, and otherwise they should work (I have added QVM and quilc docker images to the Semaphore build plan). Unfortunately, it looks like something is going on with the SCS solver, so that will need to be investigated.

### CODE REVIEW

File: forest/benchmarking/tests/test_state_tomography.py [hunk 1]
```diff
@@ -163,7 +168,7 @@ def two_q_tomo_fixture():
 
     # Get data from QVM
     tomo_expt = generate_state_tomography_experiment(state_prep, qubits)
-    results = list(measure_observables(qc=qc, tomo_experiment=tomo_expt, n_shots=100_000))
+    results = list(measure_observables(qc=qc, tomo_experiment=tomo_expt, n_shots=4000))
```
Comment (marcusps) [inline forest/benchmarking/tests/test_state_tomography.py]: Will this change require changing the tolerances in the tests that depend on these results?

Comment (karalekas) [reply forest/benchmarking/tests/test_state_tomography.py]: No, the tests all pass still. It's just that one SCS issue with the diamond norm stuff.

File: forest/benchmarking/dfe.py [hunk 2]
```diff
@@ -8,16 +8,14 @@
...
+import forest.benchmarking.operator_estimation as est
 from forest.benchmarking.utils import prepare_prod_pauli_eigenstate
 from pyquil import Program
-from pyquil.api import QuantumComputer, get_benchmarker
+from pyquil.api import BenchmarkConnection, QuantumComputer
 from pyquil.operator_estimation import ExperimentSetting, TomographyExperiment, \
     TensorProductState, plusX, minusX, plusY, minusY, plusZ, minusZ
 from pyquil.paulis import PauliTerm, PauliSum, sI, sX, sY, sZ
 
-bm = get_benchmarker()
```
Comment (kylegulshen) [inline forest/benchmarking/dfe.py]: What's wrong with initializing here? The benchmarker object doesn't serve any purpose but to perform one of two possible functions that should probably be moved directly into pyquil or benchmarking.

Comment (karalekas) [reply forest/benchmarking/dfe.py]: In addition to that, IMO it's messy to have a connection object be a module-level variable, and it is much better to explicitly declare its use in the appropriate functions.

File: forest/benchmarking/tests/conftest.py [hunk 3]
```diff
@@ -56,10 +57,10 @@ def cxn():
 @pytest.fixture(scope='session')
 def benchmarker():
     try:
-        bm = get_benchmarker()
-        bm.apply_clifford_to_pauli(Program(I(0)), sI(0))
```
Comment (kylegulshen) [inline forest/benchmarking/tests/conftest.py]: This was probably related to rigetti/pyquil#849 in case you were wondering.

Comment (karalekas) [reply forest/benchmarking/tests/conftest.py]: It definitely is related! :)

**Pull request merged**